xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/uicomponents/viewers/comments.js

```diff
@@ -87,6 +91,10 @@ viewers.Comments = Class.create({
           // without making a new request
           var comment = item.up(this.xcommentSelector);
           comment.hide();
+          const commentNbr = comment.id.match(/.+_(\d+)/)[1];
```

Comment:
I guess you're using that one to extract the number from the object name of the comment, right? Seems a bit dangerous to me, but at the very least it needs a comment IMO.